tests/jballerina-unit-test/src/test/resources/test-src/error/distinct_error_check_negative.bal

```diff
@@ -0,0 +1,40 @@
+
+type Error distinct error;
+
+type ClientError distinct Error;
+type ListenerError distinct Error;
+
+type GenericClientError distinct ClientError;
+
+type PayloadBindingClientError distinct ClientError;
+type PayloadBindingListenerError distinct ListenerError;
+type PayloadBindingError PayloadBindingClientError | PayloadBindingListenerError;
+type PayloadBindingError1 distinct ClientError | distinct ListenerError;
```

Comment:
Shall we add tests for union of error and distinct error `type PayloadBindingError2 ClientError | distinct ListenerError;` ?

Reply:
Added